src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -594,8 +595,27 @@ protected DynamoModel(IStartConfiguration config)
                 PreferenceSettings = settings;
                 PreferenceSettings.PropertyChanged += PreferenceSettings_PropertyChanged;
             }
-            // If user does not agree to GA terms, do not try to launch the client at all
-            if (PreferenceSettings.IsAnalyticsReportingApproved)
+
+            bool areAnalyticsDisabled = false;
+            try
+            {
+                // Dynamo, behind a proxy server, has been known to have issues loading the Analytics binaries.
+                // Using the "DisableAnalytics" configuration setting, a user can skip loading analytics binaries altoghether.
+                var assemblyConfig = ConfigurationManager.OpenExeConfiguration(GetType().Assembly.Location);
+                if (assemblyConfig != null)
+                {
+                    var disableAnalyticsValue = assemblyConfig.AppSettings.Settings["DisableAnalytics"];
+                    bool.TryParse(disableAnalyticsValue.Value, out areAnalyticsDisabled);
+                }
+            } catch(Exception)
+            {}
+            
+            if (areAnalyticsDisabled)
+            {
+                // If user skipped analytics from assembly configuration, do not try to launch the client at all 
+                // Skip call to instrumentation logger initialization.
+            }
+            else
```

Comment:
@QilongTang @mjkkirschner 
Turns out that we need Analytics binaries to be loaded to figure out the state of ADP opt-in/opt-out.

I am proposing we use a configuration setting to disable analytics when needed.
Opinions?

Reply:
I wouldn't do it this way, it seems this info need to be serialized, I would rather maintaining another property under preference setting which is returning `PreferenceSettings.IsAnalyticsReportingApproved|| PreferenceSettings.IsADPAnalyticsReportingApproved`. This way it can be inherited between Dynamo versions. What do you think?